src/drivers/intel/cavs/interrupt.c

```diff
@@ -52,17 +57,20 @@ static inline void irq_lvl2_handler(void *data, int level, uint32_t ilxsd,
 	struct list_item *clist;
 	uint32_t status;
 	uint32_t i = 0;
-	uint32_t unmask = 0;
 
-	/* mask the parent IRQ */
-	arch_interrupt_disable_mask(1 << level);
```

Comment:
don't mask the parent here means this handler can be re-entry with multi-instances? is that the intention?

Reply:
no, handler cannot preempt itself, handler can only be preempted by higher level IRQs